invokeai/frontend/install/import_images.py

```diff
@@ -0,0 +1,782 @@
+            elif "Dream" in img_info:
+                converted_field = parser.parse_meta_tag_dream(img_info.get("Dream"))
+            else:
+                converted_field = InvokeAIMetadata()
+                destination_needs_meta_update = False
+                print("File does not have metadata from known Invoke AI versions, add only, no update!")
+
+            # use the loaded img dimensions if the metadata didnt have them
+            if converted_field.width is None:
+                converted_field.width = png_width
+            if converted_field.height is None:
+                converted_field.height = png_height
+
+            log_version_note = converted_field.imported_app_version if converted_field else "NoVersion"
+            log_version_note = log_version_note or "NoVersion"
+
+            latest_json_string = converted_field.to_json()
+
+        print(f"From Invoke AI Version {log_version_note} with dimensions {png_width} x {png_height}.")
+
+        # if metadata needs update, then update metdata and copy in one shot
+        if destination_needs_meta_update:
+            print("Updating metadata while copying...", end="")
+            self.update_file_metadata_while_copying(
+                filepath, file_destination_path, "invokeai_metadata", latest_json_string
+            )
+            print("Done!")
+        else:
+            print("No metadata update necessary, copying only...", end="")
+            shutil.copy2(filepath, file_destination_path)
+            print("Done!")
+
+        # create thumbnail
+        print("Creating thumbnail...", end="")
+        thumbnail_path = os.path.join(config.thumbnail_path, os.path.splitext(file_name)[0]) + ".webp"
+        thumbnail_size = 256, 256
+        with PIL.Image.open(filepath) as source_image:
+            source_image.thumbnail(thumbnail_size)
+            source_image.save(thumbnail_path, "webp")
+        print("Done!")
+
+        # finalize the dynamic board name if there is an APPVERSION token in it.
+        if converted_field is not None:
+            board_name = board_name_option.replace("APPVERSION", converted_field.imported_app_version or "NoVersion")
+        else:
+            board_name = board_name_option.replace("APPVERSION", "Latest")
+
+        # maintain a map of alrady created/looked up ids to avoid DB queries
+        print("Finding/Creating board...", end="")
+        if board_name in self.board_name_id_map:
+            board_id = self.board_name_id_map[board_name]
+        else:
+            board_id = db_mapper.get_board_id_with_create(board_name)
+            self.board_name_id_map[board_name] = board_id
+        print("Done!")
+
+        # add image to db
+        print("Adding image to database......", end="")
+        modified_time = datetime.datetime.utcfromtimestamp(os.path.getmtime(filepath))
+        db_mapper.add_new_image_to_database(file_name, png_width, png_height, latest_json_string, modified_time)
```

Comment:
> Hmm, I wonder if we should be creating a new UUID for `file_name`?

My concern about creating a new uuid, is that means also renaming the files since the file name is the key. I could see the argument that if the import is a one time action for old stuff, that might be okay and preserves better consistency. That said, renaming the files means we really lose the link back and if people have separate copies in asset management systems, that might be really disruptive. Further, this import tool can be used to re-import 3.x+ files after a failed installation where a fully new install is recommended and that also feels like its more likely to cause confusion. Continually re-minting file names will also break any of the init_image links in the metadata for img2img.

If there is a dependency elsewhere that makes the old file names a problem and need to be uuid, then i recommend including the original file name in the metadata, so at least there is a trail.

Reply:
Good points. Happy to not change the filenames.